packages/cli/generation/ir-generator/src/casings/CasingsGenerator.ts

```diff
@@ -17,16 +17,23 @@ const CAPITALIZE_INITIALISM: generatorsYml.GenerationLanguage[] = ["go", "ruby"]
 
 export function constructCasingsGenerator({
     generationLanguage,
+    keywords,
     smartCasing
 }: {
     generationLanguage: generatorsYml.GenerationLanguage | undefined;
+    keywords: string[] | undefined;
     smartCasing: boolean;
 }): CasingsGenerator {
+    const keywordSet: Set<string> | undefined = keywords != null ? new Set(keywords) : undefined;
```

Comment:
nit: its probably easier to read the code if it was: 
```ts
const keywordSet: Set<string> = keywords != null ? new Set(keywords) : RESERVED_KEYWORDS[generationLanguage];
```

otherwise its less apparent that we _do_ actually use predefined reserved keywords

Reply:
Good call - it's possible `generationLanguage` is undefined at this layer, so I'll refactor this with a helper instead:

```ts
function getKeywords({
    generationLanguage,
    keywords
}: {
    generationLanguage: generatorsYml.GenerationLanguage | undefined;
    keywords: string[] | undefined;
}): Set<string> | undefined {
    if (keywords != null) {
        return new Set(keywords);
    }
    if (generationLanguage != null) {
        return RESERVED_KEYWORDS[generationLanguage];
    }
    return undefined;
}
```